File: Epic-Dining-Backend/Cafe-User-Service/src/main/java/com/example/CafeUserService/Service/CartService.java
package com.example.CafeUserService.Service;

import com.example.CafeUserService.Model.Cart;

import java.util.List;

public interface CartService {
    Cart saveCart(String userEmail, Cart cart);

    List<Cart> getCartsByUserId(String userEmail);

    boolean deleteCartItem(String userEmail, String cartId);
}

File: Epic-Dining-Frontend/src/app/services/cart.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Cart } from '../Models/cart';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class CartService {

  private baseUrl = 'http://localhost:9000/cart';

  constructor(private http: HttpClient) { }

  saveCart(userEmail: string, cart: Cart): Observable<Cart> {
    return this.http.post<Cart>(`${this.baseUrl}/save/${userEmail}`, cart);
  }

  getCartsByUserId(userEmail: string): Observable<Cart[]> {
    return this.http.get<Cart[]>(`${this.baseUrl}/get/${userEmail}`);
  }

  deleteCartItem(userEmail: string, cartId: string): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/delete/${userEmail}/${cartId}`);
  }

}
